src/bluebird.js

```diff
@@ -7,5 +7,14 @@ function noConflict() {
     return bluebird;
 }
 var bluebird = require("./promise")();
+var util = require("./util");
 bluebird.noConflict = noConflict;
 module.exports = bluebird;
+
+// Bluebird versions older then 3.1.1 (including) do not
+// support promise lifecycle events
+if (util.global.__BLUEBIRD_INSTANCES__) {
```

Comment:
Hmm, I'm not sure this would work everywhere - I recall a few environments (react native, nw.js, firefox extensions) where global detection might fail. 

For the very least this should be in a try/catch. I didn't even know we exported a util.global - that's new I guess.